src/graph/entities/attribute_set.c

```diff
@@ -212,10 +212,14 @@ void AttributeSet_Set_Allow_Null
 		return;
 	}
 
-	size_t n = ATTRIBUTESET_BYTE_SIZE(_set) + sizeof(Attribute);
-	_set = rm_realloc(_set, n);
-
 	// allocate room for new attribute
+	size_t n = ATTRIBUTESET_BYTE_SIZE(_set) + sizeof(Attribute);
+	if(set == NULL) {
```

Comment:
this condition is in conflict with the ASSERT above, should we remove that assertion ?

Reply:
fixed